### DISCUSSION TITLE
RFC: Simplify `useParam` behaviors

### DISCUSSION THREAD

**Discussion by nandorojo** [nandorojo/solito#240] [General]
I'm considering removing `initial`, `parse` and `stringify` from `useParam`. I don't like that the behavior differs across platforms.

`initial` is certainly not needed, and is confusing realistically. I think the usage should instead be like this:

```ts
// before
const [item, setItem] = useParam('item', { initial: null })

// after
const [item = null, setItem] = useParam('item')
```

`parse` should either get removed, or should apply on both web and native equally. `stringify` was only added to match react-navigation's stack, but it feels unlikely to get used. 

My hesitation with removing these is that I use them a lot. So I think this would be a sensible solution:

1. Rename `initial` to `fallback`, and apply it whenever the param is `undefined`. Currently, it uses a ref to track if it's the initial value, which feels hacky. It also might not work with concurrent mode (although, what will? lol). This approach also provides incorrect typesafety guarantees. If you set an initial value, then the TS types will not return `undefined` ever, even though they should.
2. Apply `parse` on both web and native.

**Comment by vinodsptharsha** [comment]
`const [item, setItem] = useParam('items', 'fallbackValue')`

I guess above pattern is followed by many libraries. Looks bit clean as well.
